core/src/main/scala/kafka/log/Log.scala

```diff
@@ -325,19 +325,25 @@ class Log(@volatile private var _dir: File,
     // Ensure we do not try to assign a provided topicId that is inconsistent with the ID on file.
     if (partitionMetadataFile.exists()) {
         if (!keepPartitionMetadataFile)
-          partitionMetadataFile.delete()
+          try partitionMetadataFile.delete()
+          catch {
+            case e: IOException =>
+              error(s"Error while trying to delete partition metadata file ${partitionMetadataFile}", e)
```

Comment:
Should `partitionMetadataFile` override `toString` to offer readable message?

Reply:
Good point, addressed.